dashboard/modules/serve/serve_head.py

```diff
@@ -0,0 +1,73 @@
+from aiohttp.web import Request, Response
+import json
+import logging
+
+import ray.dashboard.utils as dashboard_utils
+import ray.dashboard.optional_utils as optional_utils
+
+from ray import serve
+from ray.serve.api import deploy_group, get_deployment_statuses
+from ray.dashboard.modules.serve.schema import (
+    ServeInstanceSchema,
+    serve_instance_to_schema,
+    schema_to_serve_instance,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+routes = optional_utils.ClassMethodRouteTable
+
+
+class ServeHead(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        super().__init__(dashboard_head)
+
+    @routes.get("/api/serve/deployments/")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=True)
+    async def get_all_deployments(self, req: Request) -> Response:
+        deployments = list(serve.list_deployments().values())
+        statuses = get_deployment_statuses()
+        serve_instance_schema = serve_instance_to_schema(
+            deployments=deployments, statuses=statuses
+        )
+        return Response(
+            text=json.dumps(serve_instance_schema.json()),
+            content_type="application/json",
+        )
```

Comment:
this came up in design review; let's separate getting the goal state/config of deployments w/ getting the current status (concretely, split this into a `GET /api/serve/deployments/status` API

this makes it simpler from the caller side: what you `PUT` for the config is exactly what you `GET`

Reply:
Good call, I updated the API with a separate status endpoint.